Libraries/Text/TextInput/Singleline/RCTUITextField.m

```diff
@@ -176,6 +176,16 @@ - (CGRect)editingRectForBounds:(CGRect)bounds
 
 #pragma mark - Overrides
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-implementations"
```

Comment:
Hmm, why do we need these pragmas? This seems like a sign that this might not be the right solution.

Reply:
Yeah, we need it to suppress warning, warning because we mark `setSelectedTextRange ` as unavailable, please see : https://github.com/facebook/react-native/blob/7a7eb11965a4126987fd0f4bfcd13dd872452e76/Libraries/Text/TextInput/RCTBackedTextInputViewProtocol.h#L30-L36

We disallow access `setSelectedTextRange ` directly, but Apple framework would call it, and this is how KVO works.